CRM/Core/EntityTokens.php

```diff
@@ -176,78 +198,56 @@ protected function getFieldMetadata(): array {
         $this->fieldMetadata = [];
       }
     }
-    return $this->fieldMetadata;
+    if ($field) {
+      return $this->fieldMetadata[$field] ?? NULL;
+    }
+    else {
+      return $this->fieldMetadata;
+    }
   }
 
   /**
    * Get pseudoTokens - it tokens that reflect the name or label of a pseudoconstant.
    *
-   * @internal - this function will likely be made protected soon.
-   *
+   * @internal - this function is a bridge for legacy CRM_Utils_Token callers. It should be removed.
```

Comment:
We don't need to deprecate this - we can move or remove it - none of the functions on this class are set in stone - they are only used in recent tested core places

Having said that - the point of this was to take the metadata for the entity and add the pseudotokens where the metadata required it - once we remove the hard coded list of tokens in favour of metadata we will need 'something' to add in these tokens